src/routes/concepts/components/event-handlers.mdx

```diff
@@ -3,68 +3,209 @@ title: Event Handlers
+
+```tsx
+<div onClick={() => props.handleClick?.()} />
+```
 
 ## Event delegation
 
-Instead of attaching event listeners to every individual element, Solid uses artificial event delegation.
-This means that event listeners are attached to the root element and events are captured as they bubble up.
+Instead of attaching event listeners to every individual element, Solid uses (via the regular [`on__`](/reference/jsx-attributes/on_) form), what's known as _synthetic event delegation_.
+This means that event listeners are attached to the `document` element, and dispatched to the relevant elements as the event bubbles up.
 
 By keeping the number of event listeners to a minimum, events can be captured more effectively.
-This is especially useful when working with large lists of elements, such as in a table or list.
+This is especially useful when working with a large number of elements, such as in a table or list.
 
-Supported events such as `click`, `mouseover`, and `keydown` are just a few examples that are optimized in this way.
+Supported events such as `click`, `input` and `keydown` are just a few examples that are optimized in this way.
 To view the full list see the [references below](#delegated-events).
 
-If you need to attach an event listener to an element that is not supported by Solid's event delegation, you can use the [on:](/reference/jsx-attributes/on-and-oncapture) prefix followed by the event name, and assign it a function.
+If you need to attach an event listener to an element that is not supported by Solid's event delegation, such a custom event in a [Custom Element](https://developer.mozilla.org/en-US/docs/Web/API/Web_components/Using_custom_elements), you can use the [`on:__`](/reference/jsx-attributes/on-and-oncapture) form.
 
 ```tsx
 <div on:customEvent={handleCustomEvent} />
 ```
 
-## Binding events
+### Event delegation considerations
+
+While delegated events provide some performance enhancements, it comes with some tradeoffs that should be considered.
+
+<Callout type="info" title="Design">
 
-In Solid, event handlers can be bound by passing an array to the event handler.
-The second item of the array is supplied as the handler's first argument.
+    Event delegation is designed for event propagation through the JSX Tree, rather than the DOM Tree.
+
+</Callout>
+
+This differs from the acquired expectations one may have over how events work and flow.
+
+Here is a list of peculiarities to keep in mind:
+
+1. Delegated event listeners are added to the `document` once per event type.
+   When the element/component that added it is removed, it's handler function will no longer be called, but the delegated event listener will still be present on the `document`.
+   Considering the following example, a `div` listening for `mousemove`.
+   `mousemove` will still be dispatched to the `document` even if the `div` is disposed.
```

Comment:
```suggestion
Event delegation is designed for event propagation through the JSX Tree, rather than the DOM Tree.
This can differ from the previous expectations of how events work and flow.
Some things to keep in mind include:
- Delegated event listeners remain active even if the element that added them is removed. 
For example, if a `div` listens for `mousemove` and is later removed, the `mousemove` events will still be dispatched to the `document`. 
What this means is that the listener persists even after the originating element or component is disposed.
```

Reply:
@LadyBluenotes I think these lines

> Delegated event listeners are added to the `document` once per event type. When the element/component that added it is removed, it's handler function will no longer be called, but the delegated event listener will still be present on the `document`.

are useful for understanding how it behaves. They give a bit more context as to why it will still remain even once the element that added it is removed, and gives some clarity that it is the delegated event listener only and not the handler that created it will persist. I actually think they way I wrote this wasn't enough and kinda wish there was even more context. I feel like just saying the delegated event listener persists doesn't paint a clear picture of how and why it behaves this way.

Maybe this wording could be used instead as a compromise? 

```suggestion
Event delegation is designed for event propagation through the JSX Tree, rather than the DOM Tree.
This can differ from the previous expectations of how events work and flow.
Some things to keep in mind include:
 - Delegated event listeners remain active even if the element that added them is removed. 
For example, if a `div` listens for `mousemove` and is later removed, the `mousemove` events will still be dispatched to the `document`. 
Delegated event listeners are added _once_ per event type and handle all events of that type. Once setup, the delegated event listener will call the handlers attached to any existing elements, but the listener itself will persist even if all elements with that handler type are removed.
```

cc @titoBouzout

Reply:
ok, I will merge both suggestions but edit the last sentence because it's slightly repetitive.  I will edit locally and post a new comment